src/com/nisovin/magicspells/spells/passive/GiveDamageListener.java

```diff
@@ -32,30 +33,60 @@
 	Set<Material> types = new HashSet<>();
 	Map<MagicMaterial, List<PassiveSpell>> weapons = new LinkedHashMap<>();
 	List<PassiveSpell> always = new ArrayList<>();
+	boolean ignoreSpellDamage = false;
+	boolean physicalOnly = false;
 	
 	@Override
 	public void registerSpell(PassiveSpell spell, PassiveTrigger trigger, String var) {
 		if (var == null || var.isEmpty()) {
 			always.add(spell);
 		} else {
-			String[] split = var.split(",");
-			for (String s : split) {
-				s = s.trim();
-				MagicMaterial mat;
-				if (s.contains("|")) {
-					String[] stuff = s.split("\\|");
-					mat = MagicSpells.getItemNameResolver().resolveItem(stuff[0]);
-					if (mat != null) mat = new MagicItemWithNameMaterial(mat, stuff[1]);						
-				} else {
-					mat = MagicSpells.getItemNameResolver().resolveItem(s);
+			String[] arguments = var.split(" ");
+			for (String argument : arguments) {
+				if (argument.equalsIgnoreCase("ignoreSpellDamage")) {
+					ignoreSpellDamage = true;
+					continue;
 				}
-				if (mat != null) {
-					List<PassiveSpell> list = weapons.computeIfAbsent(mat, magicMaterial -> new ArrayList<>());
-					list.add(spell);
-					types.add(mat.getMaterial());
+				
+				if (argument.equalsIgnoreCase("physicalOnly")) {
+					physicalOnly = true;
+					continue;
 				}
+				
+				String[] split = var.split(",");
+				for (String s : split) {
+					s = s.trim();
+					MagicMaterial mat;
+					if (s.contains("|")) {
+						String[] stuff = s.split("\\|");
+						mat = MagicSpells.getItemNameResolver().resolveItem(stuff[0]);
+						if (mat != null) mat = new MagicItemWithNameMaterial(mat, stuff[1]);						
+					} else {
+						mat = MagicSpells.getItemNameResolver().resolveItem(s);
+					}
+					if (mat != null) {
+						List<PassiveSpell> list = weapons.computeIfAbsent(mat, magicMaterial -> new ArrayList<>());
+						list.add(spell);
+						types.add(mat.getMaterial());
+					}
+				}
+			}
+		}
+	}
+	
+	public static boolean startedByMagicSpells() {
+		int ctr = 0;
+		for (StackTraceElement element : Thread.currentThread().getStackTrace()) {
+			if (ctr > 30) {
+				break;
+			}
+			String pack = element.getClassName();
+			if (pack.contains("com.nisovin.magicspells.") && !pack.contains("com.nisovin.magicspells.MagicSpells")) {
```

Comment:
Won't this always cause it to return true since it is checking for a pattern in the package in which the class this exists in will match? Meaning that this method being called would cause it to return true?

Reply:
Your right pushing a commit to fix this now, was porting this from a separate plugin I had fixing this.

Reply:
Bad port logic has been changed. I just got done fully testing this, kicked off that merge a bit too soon, figured it would work out the gate.